src/builtins/map.js

```diff
@@ -78,16 +78,19 @@ export default function map ( inputdir, outputdir, options ) {
 								return reject( err );
 							}
 
-							if ( result === null ) return fulfil();
+							return Promise.resolve( result )
+								.then( result => {
+									if ( result === null ) return fulfil();
 
-							const codepath = resolve( this.cachedir, filename );
+									const codepath = resolve( this.cachedir, filename );
 
-							const { code, map } = processResult( result, data, src, dest, codepath );
+									const { code, map } = processResult( result, data, src, dest, codepath );
 
-							writeToCacheDir( code, map, codepath, dest )
```

Comment:
Shouldn't be this promise chain be `return`ed so if there's an error it'll propagate? I copied the current implementation, but I think it might be an error.